services/sqs/src/main/java/software/amazon/awssdk/services/sqs/batchmanager/BatchOverrideConfiguration.java

```diff
@@ -16,169 +16,133 @@
- * are not specified.
+ * Configuration values for the BatchManager implementation used for controlling batch operations.
+ * All values are optional, and default values will be used if they are not specified.
  */
 @SdkPublicApi
 public final class BatchOverrideConfiguration implements ToCopyableBuilder<BatchOverrideConfiguration.Builder,
     BatchOverrideConfiguration> {
 
-    private final Integer maxBatchItems;
-    private final Integer maxBatchKeys;
-    private final Integer maxBufferSize;
-    private final Duration maxBatchOpenDuration;
-    private final Duration visibilityTimeout;
-    private final Duration longPollWaitTimeout;
-    private final Duration minReceiveWaitTime;
-    private final Integer maxDoneReceiveBatches;
-    private final List<MessageSystemAttributeName> messageSystemAttributeNames;
+    private final Integer maxBatchSize;
+    private final Duration sendMessageFrequency;
+    private final Duration receiveMessageVisibilityTimeout;
+    private final Duration receiveMessageMinWaitDuration;
+    private final List<MessageSystemAttributeName> receiveMessageSystemAttributeNames;
     private final List<String> receiveMessageAttributeNames;
-    private final Boolean adaptivePrefetching;
-    private final Integer maxInflightReceiveBatches;
+
 
     private BatchOverrideConfiguration(Builder builder) {
-        this.maxBatchItems = Validate.isPositiveOrNull(builder.maxBatchItems, "maxBatchItems");
-        this.maxBatchKeys = Validate.isPositiveOrNull(builder.maxBatchKeys, "maxBatchKeys");
-        this.maxBufferSize = Validate.isPositiveOrNull(builder.maxBufferSize, "maxBufferSize");
-        this.maxBatchOpenDuration = Validate.isPositiveOrNull(builder.maxBatchOpenDuration, "maxBatchOpenDuration");
-        this.visibilityTimeout = Validate.isPositiveOrNull(builder.visibilityTimeout, "visibilityTimeout");
-        this.longPollWaitTimeout = Validate.isPositiveOrNull(builder.longPollWaitTimeout, "longPollWaitTimeout");
-        this.minReceiveWaitTime = Validate.isPositiveOrNull(builder.minReceiveWaitTime, "minReceiveWaitTime");
-        this.messageSystemAttributeNames = builder.messageSystemAttributeNames;
-        this.receiveMessageAttributeNames = builder.receiveMessageAttributeNames;
-        this.adaptivePrefetching = builder.adaptivePrefetching;
-        this.maxInflightReceiveBatches = builder.maxInflightReceiveBatches;
-        this.maxDoneReceiveBatches = builder.maxDoneReceiveBatches;
+        this.maxBatchSize = Validate.isPositiveOrNull(builder.maxBatchSize,
+                                                                "maxBatchSize");
+        Validate.isTrue(this.maxBatchSize == null || this.maxBatchSize <= 10,
+                        "The maxBatchSize must be less than or equal to 10. A batch can contain up to 10 messages.");
+
+        this.sendMessageFrequency = Validate.isPositiveOrNull(builder.sendMessageFrequency,
```

Comment:
I understand this is most voted, but should we do `sendFrequency` or `sendRequestFrequency`? My concern is that people may confuse it to be only associated with SendMessageRequest.